crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -129,13 +129,13 @@ impl Trade {
     ///
     /// Denominated in NATIVE token
     fn native_surplus(&self, prices: &auction::Prices) -> Result<eth::Ether, Error> {
-        let surplus = self.surplus_token_price(prices)?.apply(
-            self.surplus()
-                .ok_or(Error::Surplus(self.sell, self.buy))?
-                .amount,
-        );
+        let surplus = self.surplus().ok_or(Error::Surplus(self.sell, self.buy))?;
+        let price = prices
+            .get(&surplus.token)
+            .ok_or(Error::MissingPrice(surplus.token))?;
+
         // normalize
-        Ok((surplus.0 / *UNIT).into())
+        Ok((price.apply(surplus.amount).0 / *UNIT).into())
```

Comment:
Can we reproduce the behavior we have in the solvers crate (ie expose an `in_eth` method on `Price` or ideally even on TokenAmount) and return this instead of redefining UNIT here? I don't find it very intuitive what applying a price to an ammount should mean.

Also, if `apply` already returns `eth::Ether`, why do we have to divide by `UNIT` again here?

https://github.com/cowprotocol/services/blob/79ec9d9d056811c1d45eeeec878a813c0036ffae/crates/solvers/src/domain/auction.rs#L69-L83

Reply:
I believe this is the right fix: https://github.com/cowprotocol/services/pull/2499

Reply:
That said, is there actually a bug in the attached example (?):

`eth.0.checked_mul(Self::BASE.into())?.checked_div(self.0 .0) `

Reply:
Fix is merged, resolving